src/azul/transformer.py

```diff
@@ -125,7 +125,16 @@ def translate_fields(cls, source: JSON, path: Tuple[str, ...] = (), forward: boo
         :return: A copy of the original document with values translated according to their type
         """
         if isinstance(source, dict):
-            return {key: cls.translate_fields(val, path=path + (key,), forward=forward) for key, val in source.items()}
+            new_dict = {}
+            for key, val in source.items():
+                # Skip any shadow copy fields to remove them from the dictionary when reverse translating
+                if not key.endswith('_'):
```

Comment:
```suggestion
                if key.endswith('_'):
                    assert not forward
                else:
```

The usual strictness rule.